src/common/fetch/Fetch.ts

```diff
@@ -0,0 +1,81 @@
+import axios, { AxiosInstance } from 'axios';
+
+import {
+  Config,
+  IFetch,
+  RejectedCallback,
+  ExtendedAxiosRequestConfig,
+  RequestInterceptor,
+  RequestSuccessCallback,
+  ResponseInterceptor,
+  ResponseSuccessCallback,
+} from './types';
+
+export class Fetch implements IFetch {
+  private axios: AxiosInstance;
+
+  constructor({ config, requestInterceptors = [], responseInterceptors = [] }: Config = {}) {
+    this.axios = axios.create(config);
+
+    requestInterceptors?.forEach((interceptor) => {
+      this.registerRequestInterceptor(interceptor);
+    });
+
+    responseInterceptors?.forEach((interceptor) => {
+      this.registerResponseInterceptor(interceptor);
+    });
+  }
+
+  registerRequestInterceptor({ onSuccess, onError }: RequestInterceptor) {
+    this.axios.interceptors.request.use(onSuccess as RequestSuccessCallback, onError as RejectedCallback);
+  }
+
+  registerResponseInterceptor({ onSuccess, onError }: ResponseInterceptor) {
+    this.axios.interceptors.response.use(onSuccess as ResponseSuccessCallback, onError as RejectedCallback);
+  }
+
+  request({ onSuccess, onError, ...request }: ExtendedAxiosRequestConfig) {
+    const controller = new AbortController();
+
+    const responsePromise = this.axios
+      .request({ ...request, signal: controller.signal } as ExtendedAxiosRequestConfig)
+      .then((data) => {
+        // простой способ обработки запроса через коллбеки + сохраняется возможность обрабатывать Promise и по месту вызова
+        return Promise.resolve(onSuccess ? onSuccess(data) : data);
+      })
+      .catch((error) => {
+        return Promise.reject(onError ? onError(error) : error);
+      });
+
+    return {
+      response: responsePromise,
+      abort: () => {
+        controller.abort();
+      },
+    };
+  }
+
+  get(request: Omit<ExtendedAxiosRequestConfig, 'method'>) {
+    return this.request({ ...request, method: 'GET' });
```

Comment:
I think we can use constants METHODS for method: 'GET'

Reply:
Y’r absolutely right